plugins/chat/assets/javascripts/discourse/controllers/chat.js

```diff
@@ -46,4 +58,16 @@ export default class ChatController extends Controller {
 
     return modifierClasses.join(" ");
   }
+
+  @action
+  onClickTab(tab) {
+    switch (tab) {
+      case "threads":
+        return this.router.transitionTo("chat.threads");
+      case "direct-messages":
+        return this.router.transitionTo("chat.direct-messages");
+      case "channels":
+        return this.router.transitionTo("chat.channels");
+    }
```

Comment:
How about this one-liner?

```suggestion
    return this.router.transitionTo(`chat.${tab}`);
```